wordgenerator.py

```diff
@@ -0,0 +1,110 @@
+        for x in range (self.smallest_length, self.number_of_lengths):
+            self.index_locations.append(self.locate_first_length(x))
+        self.index_prime_locations = deepcopy(self.index_locations)
+
+    '''
+    log(n) binary search to find the first length
+    '''
+    def locate_first_length(self, requested_length):
+        found = False
+        index = math.ceil(len(self.word_array)/2)
+        delta = math.ceil(len(self.word_array)/4)
+        while True:
+            if self.check_if_first(index, requested_length):
+                break
+            if len(self.word_array[index]) >= requested_length:
+                index -= delta
+            else:
+                index += delta 
+            if index < 0:
+                index = 0
+            delta = math.ceil(delta/2)
+        return index
+
+    def check_if_first(self, index, requested_length):
+        if len(self.word_array[index]) != requested_length:
+            return False
+        if index == 0:
+            return True
+        return len(self.word_array[index]) != len(self.word_array[index-1])
+
+    '''
+    If you want words with length 4-6 (inclusive), request_word(4,3)
+    If you want words with length of 2, request_word(2,1)
+    '''
+    def request_word(self, index, number_of_words):
+        if (index + number_of_words) > self.number_of_lengths: 
+            print("WordGenerator: ERROR: Range you requested was too large, giving you the largest valid range instead.")
+            number_of_words = self.number_of_lengths - index
+        if (index >= self.number_of_lengths):
+            print("WordGenerator: ERROR: Index you requested was too large, giving you the largest index instead.")
+            index = self.number_of_lengths - 1
+            number_of_words = 1
+        if number_of_words >= 1:
+            index = random.randrange(index, index + number_of_words)
+        index_of_word = self.index_prime_locations[index]
+        self.index_prime_locations[index] += 1
+        if index >= len(self.index_locations)-1:
+            if self.index_prime_locations[index] == (len(self.word_array) - 1):
+                print("You have exhuasted the pool of length " + repr(index) + " words, looping back to beginning")
+                self.index_prime_locations[index] = self.index_locations[index]
+        elif self.index_prime_locations[index] >= self.index_locations[index+1]:
+            print("You have exhuasted the pool of length " + repr(index) + " words, looping back to beginning")
+            self.index_prime_locations[index] = self.index_locations[index] 
+        return self.word_array[index_of_word]
+
+    def print_list(self):
+        for i in range(0, len(self.word_array)):
+            print("Index: " + repr(i) + " Word: " + self.word_array[i])
+
+
+if __name__ == '__main__':
+    tester = WordGenerator()
+    for i in range(0,100):
+        print("random word: " + tester.request_word(19,1))
+
+
```

Comment:
Should remove the extra white lines at the end here, small nitpicky thing. I just finished fixing whitespace in the other files so it would be nice if this one conformed also.